project/Build.scala

```diff
@@ -1363,66 +1344,65 @@ object Build {
+
       generateScalaDocumentation := Def.inputTaskDyn {
-        val extraArgs = spaceDelimited("[output]").parsed
-        val dest = file(extraArgs.headOption.getOrElse("scaladoc/output/scala3")).getAbsoluteFile
-        val justAPI = extraArgs.drop(1).headOption == Some("--justAPI")
         val majorVersion = (LocalProject("scala3-library-bootstrapped") / scalaBinaryVersion).value
-        val dottyJars: Seq[java.io.File] = Seq(
-          (`stdlib-bootstrapped`/Compile/products).value,
-          (`scala3-interfaces`/Compile/products).value,
-          (`tasty-core-bootstrapped`/Compile/products).value,
-        ).flatten
-
-        val roots = dottyJars.map(_.getAbsolutePath)
-
-        val managedSources =
-          (`stdlib-bootstrapped`/Compile/sourceManaged).value / "scala-library-src"
-        val projectRoot = (ThisBuild/baseDirectory).value.toPath
-        val stdLibRoot = projectRoot.relativize(managedSources.toPath.normalize())
-        val docRootFile = stdLibRoot.resolve("rootdoc.txt")
-
-        val dottyManagesSources =
-          (`stdlib-bootstrapped`/Compile/sourceManaged).value / "dotty-library-src"
-
-        val dottyLibRoot = projectRoot.relativize(dottyManagesSources.toPath.normalize())
-
-        def generateDocTask =
-          generateDocumentation(
-            roots, "Scala 3", dest.getAbsolutePath, "main",
-            Seq(
-              "-comment-syntax", "wiki",
-              s"-source-links:docs=github://lampepfl/dotty/main#docs",
-              "-doc-root-content", docRootFile.toString,
-              "-versions-dictionary-url",
-              "https://scala-lang.org/api/versions.json",
-              "-Ydocument-synthetic-types",
-              s"-snippet-compiler:${dottyLibRoot}/scala/quoted=compile,${dottyLibRoot}/scala/compiletime=compile"
-            ) ++ (if (justAPI) Nil else Seq("-siteroot", "docs", "-Yapi-subdirectory")), includeExternalMappings = false)
-
-        if (dottyJars.isEmpty) Def.task { streams.value.log.error("Dotty lib wasn't found") }
-        else if (justAPI) generateDocTask
-        else Def.task{
-          IO.write(dest / "versions" / "latest-nightly-base", majorVersion)
-
-          // This file is used by GitHub Pages when the page is available in a custom domain
-          IO.write(dest / "CNAME", "dotty.epfl.ch")
-        }.dependsOn(generateDocTask)
+
+        val extraArgs = spaceDelimited("[<output-dir>] [--justAPI]").parsed
+        val outputDirOverride = extraArgs.headOption.fold(identity[GenerationConfig](_))(newDir => {
+          config: GenerationConfig => config.add(OutputDir(newDir))
+        })
+        val justAPIArg: Option[String] = extraArgs.drop(1).find(_ == "--justAPI")
+        val justAPI = justAPIArg.fold(identity[GenerationConfig](_))(_ => {
+          config: GenerationConfig => config.remove[SiteRoot]
+        })
+        val overrideFunc = outputDirOverride.andThen(justAPI)
+
+        val config = Def.task {
+          overrideFunc(Scala3.value)
```

Comment:
NIT: Just using Scala 3 here looks strange, I would prefix it with `ScaladocSettings.Scala3`